### PR TITLE
Document different handling of archive files with the wasm backend

### PR DESCRIPTION
**Pull request by sbc100**
See https://github.com/emscripten-core/emscripten/issues/9705

### CODE REVIEW

File: site/source/docs/compiling/WebAssembly.rst [hunk 1]
```diff
@@ -56,6 +65,14 @@ There are some differences you may notice between the two backends, if you upgra
     LLVM optimization passes by default, even if using LLVM IR in object files;
     for that you must pass ``--llvm-lto 1``.
 
+* `wasm-ld`, the linker used by the wasm backend, requires librarys (`.a`
```
Comment (dschuff) [inline site/source/docs/compiling/WebAssembly.rst]: "libraries"

File: site/source/docs/compiling/WebAssembly.rst [hunk 2]
```diff
@@ -56,6 +65,14 @@ There are some differences you may notice between the two backends, if you upgra
...
+  archives) to continue symbol indexes.  This matches the behaviour the native
```
Comment (dschuff) [inline site/source/docs/compiling/WebAssembly.rst]: continue -> contain?

File: site/source/docs/compiling/WebAssembly.rst [hunk 3]
```diff
@@ -56,6 +65,14 @@ There are some differences you may notice between the two backends, if you upgra
...
+  GNU linker.  While `emar` will create such indexes by default, native tools
+  such as GNU `ar` and GNU `strip` are not aware of the WebAssembly object
+  format and cannot create archive indexes.  In particular if you run GNU
+  `strip` on an archive file contains WebAssembly abject files it will remove
```
Comment (dschuff) [inline site/source/docs/compiling/WebAssembly.rst]: "archive file that contains", and abject->object

Comment (sbc100) [reply site/source/docs/compiling/WebAssembly.rst]: Right.. i wouldn't want to pass judgement on this miserable abject files!

**Pull request merged**